oauth2_http/javatests/com/google/auth/oauth2/functional/FTComputeEngineCredentialsTest.java

```diff
@@ -0,0 +1,86 @@
+package com.google.auth.oauth2.functional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.client.json.webtoken.JsonWebSignature;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.ComputeEngineCredentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.IdToken;
+import com.google.auth.oauth2.IdTokenCredentials;
+import com.google.auth.oauth2.IdTokenProvider;
+import org.junit.Test;
+
+public final class FTComputeEngineCredentialsTest {
+
+  private final String cloudTasksUrl =
+      "https://cloudtasks.googleapis.com/v2/projects/gcloud-devel/locations";
+  private final String cloudPlatformScope = "https://www.googleapis.com/auth/cloud-platform";
+
+  @Test
+  public void RefreshCredentials() throws Exception {
+
+    ComputeEngineCredentials credentials = ComputeEngineCredentials.create();
+
+    credentials.refresh();
+
+    assertNotNull(credentials.getAccessToken());
+    assertNotNull(credentials.getAccount());
+  }
+
+  @Test
+  public void DefaultCredentials() throws Exception {
+    ComputeEngineCredentials computeEngineCredential = ComputeEngineCredentials.create();
```

Comment:
to make this an actual ADC test - you want to create credentials from GoogleCredential.getApplicationDefault()

Reply:
This might not work if there is another ADC set in the GH environment, but lets try.

Reply:
Done.